src/rules/__tests__/prefer-importing-jest-globals.test.ts

```diff
@@ -0,0 +1,451 @@
+    `,
+    errors: [
+      {
+        endColumn: 9,
+        column: 1,
+        line: 1,
+        messageId: 'preferImportingJestGlobal',
+      } as const,
+    ],
+  },
+];
+
+ruleTester.run('prefer-importing-jest-globals', rule, {
+  valid: [
+    // {
+    //   code: dedent`
+    //     // with import
+    //     import { test, expect } from '@jest/globals';
+    //     test('should pass', () => {
+    //         expect(true).toBeDefined();
+    //     });
+    //     `,
+    //   parserOptions: { sourceType: 'module' },
+    // },
+    // {
+    //   code: dedent`
+    //     // with require
+    //     const { test, expect } = require('@jest/globals');
+    //     test('should pass', () => {
+    //         expect(true).toBeDefined();
+    //     });
+    //     `,
+    // },
+    // {
+    //   code: dedent`
+    //     import { it as itChecks } from '@jest/globals';
+    //     itChecks("foo");
+    //   `,
+    //   parserOptions: { sourceType: 'module' },
+    // },
+    // {
+    //   code: dedent`
+    //     const { test } = require('@jest/globals');
+    //     test("foo");
+    //   `,
+    // },
+    // {
+    //   code: dedent`
+    //     const { test } = require('my-test-library');
+    //     test("foo");
+    //   `,
+    // },
+  ],
+  invalid: [
+    ...importImports,
+    ...requireImports,
+    {
+      code: dedent`
+        #!/usr/bin/env node
+        describe("suite", () => {
+          test("foo");
+          expect(true).toBeDefined();
+        })
+      `,
+      output: dedent`
+        #!/usr/bin/env node
+        const { describe, test, expect } = require('@jest/globals');
+        describe("suite", () => {
+          test("foo");
+          expect(true).toBeDefined();
+        })
+    `,
+      parserOptions: { sourceType: 'script' },
+      errors: [
+        {
+          endColumn: 9,
+          column: 1,
+          line: 2,
+          messageId: 'preferImportingJestGlobal',
+        },
+      ],
+    },
+    {
+      code: dedent`
+        // with comment above
+        describe("suite", () => {
+          test("foo");
+          expect(true).toBeDefined();
+        })
+      `,
+      output: dedent`
+        // with comment above
+        const { describe, test, expect } = require('@jest/globals');
+        describe("suite", () => {
+          test("foo");
+          expect(true).toBeDefined();
+        })
+    `,
+      errors: [
+        {
+          endColumn: 9,
+          column: 1,
+          line: 2,
+          messageId: 'preferImportingJestGlobal',
+        },
+      ],
+    },
+    {
+      code: dedent`
+        'use strict';
+        describe("suite", () => {
+          test("foo");
+          expect(true).toBeDefined();
+        })
+      `,
+      output: dedent`
+        'use strict';
+        const { describe, test, expect } = require('@jest/globals');
+        describe("suite", () => {
+          test("foo");
+          expect(true).toBeDefined();
+        })
+    `,
+      errors: [
```

Comment:
nit: ideally these should align, but I'm happy to take care of that in post:

```suggestion
        })
      `,
      errors: [
```

Reply:
Prettier took care of it:
![image](https://github.com/jest-community/eslint-plugin-jest/assets/1864359/f2c09180-dc07-4f70-9d33-f1119ceee42e)